### ISSUE TITLE
What happen with `prisma migrate dev` 

### ISSUE THREAD

**Issue by und3fined** [prisma/prisma#16220]
### Bug description

Hi team,

I use command `prisma migrate dev --skip-generate --schema $npm_package_pschema` for update new change in table.
I have a problem with before migrate. Accept your warning first time, and re-try `migrate dev` it's comback.
After 4 times re-try, i have 4 migrate change with same content. 

This here.
<img width="1181" alt="Screenshot 2022-11-10 at 17 11 39" src="https://user-images.githubusercontent.com/2163878/201063341-c1c2dd0f-d4f1-4e11-86f8-14db77c77cf4.png">

No any change but always drop - and create new field. Same type with `UserRole`. but it's not show in here.

### How to reproduce

`prisma migrate dev --skip-generate --schema $npm_package_pschema`

And always got 

```
Warnings for the current datasource:

  • The `state` column on the `app_versions` table would be dropped and recreated. This will lead to data loss.
  • The `rank` column on the `users` table would be dropped and recreated. This will lead to data loss.

✔ Are you sure you want create and apply this migration? … no
```

### Expected behavior

Don't DROP field

### Prisma information

```prisma

generator client {
    provider = "prisma-client-js"
}

datasource db {
    provider          = "postgresql"
    url               = env("DB_URL")
    shadowDatabaseUrl = env("SD_DB_URL")
}

enum VersionState {
    NEW
    PRE_RELEASE
    READY
}

model AppVersion {
    id String @id @default(cuid()) @db.VarChar(48)

    version      String       @db.VarChar(255)
    state        VersionState @default(NEW)
}

enum UserRole {
    GUEST
    MEMBER
    SILVER
    PLATINUM
    GOLD
    DIAMOND

    DECRYPTOR
    MODERATOR
    ADMIN
}

enum UserRank {
    NEWBIE
    MEMBER
    SILVER
    PLATINUM
    GOLD
    DIAMOND
}

model User {
    id        String @id @default(cuid()) @db.VarChar(48)
    profileId String @unique @map("profile_id") @db.VarChar(48)

    displayName String   @default("Anonymous") @map("display_name") @db.VarChar(255)
    role        UserRole @default(GUEST)
    rank        UserRank @default(NEWBIE)
}

```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: macOS M1
- Database: Postgresql 14
- Node.js version: v16.14.2
 

### Prisma Version

```
prisma                  : 4.6.0
@prisma/client          : 4.6.0
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine 2e719efb80b56a3f32d18a62489de95bb9c130e3 (at node_modules/.pnpm/@prisma+engines@4.6.0/node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli 2e719efb80b56a3f32d18a62489de95bb9c130e3 (at node_modules/.pnpm/@prisma+engines@4.6.0/node_modules/@prisma/engines/migration-engine-darwin-arm64)
Introspection Engine    : introspection-core 2e719efb80b56a3f32d18a62489de95bb9c130e3 (at node_modules/.pnpm/@prisma+engines@4.6.0/node_modules/@prisma/engines/introspection-engine-darwin-arm64)
Format Binary           : prisma-fmt 2e719efb80b56a3f32d18a62489de95bb9c130e3 (at node_modules/.pnpm/@prisma+engines@4.6.0/node_modules/@prisma/engines/prisma-fmt-darwin-arm64)
Format Wasm             : @prisma/prisma-fmt-wasm 4.6.0-53.2e719efb80b56a3f32d18a62489de95bb9c130e3
Default Engines Hash    : 2e719efb80b56a3f32d18a62489de95bb9c130e3
Studio                  : 0.476.0
```

**Comment by Jolg42** [comment]
Confirmed as duplicate of #16180.

@und3fined Could you try to update `prisma` and `@prisma/client` to the `4.6.1-dev.1` internal patch version that includes the fixes and let us know if that solves your issue?

The official patch version will be published later today.

**Comment by und3fined** [comment]
It's fixed in 4.6.1 thanks for quickly reply
